core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -886,12 +886,24 @@ protected Scope visitAnalyze(Analyze node, Optional<Scope> scope)
 
             analysis.setAnalyzeMetadata(metadata.getStatisticsCollectionMetadata(session, tableHandle, analyzeProperties));
 
+            Set<Field> fields = metadata.getTableSchema(session, tableHandle)
+                    .columns().stream()
+                    .map(column -> Field.newQualified(
+                            node.getTableName(),
+                            Optional.of(column.getName()),
+                            column.getType(),
+                            column.isHidden(),
+                            Optional.of(tableName),
+                            Optional.of(column.getName()),
+                            false))
+                    .collect(Collectors.toSet());
```

Comment:
In general, that PR is what I started doing, but this is the place where I got into trouble, because I didn't know where to get a `Field` instance that I would put into the collection of referenced columns.